corrector_module/opmon_corrector/corrector_batch.py

```diff
@@ -91,40 +117,79 @@ def _batch_run(self, process_dict):
         m = multiprocessing.Manager()
         to_remove_queue = m.Queue()
 
-        for message_id in doc_map:
-            documents = doc_map[message_id]
+        for x_request_id in doc_map:
+            documents = doc_map[x_request_id]
             data = dict()
             data['logger_manager'] = self.logger_m
             data['document_manager'] = doc_m
-            data['message_id'] = message_id
+            data['x_request_id'] = x_request_id
             data['documents'] = documents
             data['to_remove_queue'] = to_remove_queue
             list_to_process.put(data)
             doc_len += len(documents)
 
-        # Sync
-        # time.sleep(5)
+        self._process_workers(list_to_process, duplicates)
 
-        # Create pool of workers
-        pool = []
-        for i in range(self.settings["corrector"]["thread-count"]):
-            # Configure worker
-            worker = CorrectorWorker(self.settings, f'worker_{i}')
-            p = multiprocessing.Process(target=worker.run, args=(list_to_process, duplicates))
-            pool.append(p)
+        # Go through the to_remove list and remove the duplicates
+        element_in_queue = True
+        total_raw_removed = 0
+        while element_in_queue:
+            try:
+                # Do not block queue and return element immediately else raise the queue.Empty
+                element = to_remove_queue.get(block=False)
+                db_m.remove_duplicate_from_raw(element)
+                total_raw_removed += 1
+            except queue.Empty:
+                element_in_queue = False
+        if total_raw_removed > 0:
+            self.logger_m.log_info('corrector_batch_remove_duplicates_from_raw',
+                                   'Total of {0} duplicate documents removed from raw messages.'.format(
+                                       total_raw_removed))
+        else:
+            self.logger_m.log_info('corrector_batch_remove_duplicates_from_raw',
+                                   'No raw documents marked to removal.')
 
-        # Starts all pool process
-        for p in pool:
-            p.start()
-        # Wait all processes to finish their jobs
-        for p in pool:
-            p.join()
+        doc_len += total_raw_removed
+
+        end_processing_time = time.time()
+        total_time = time.strftime(
+            PROCESSING_TIME_FORMAT,
+            time.gmtime(end_processing_time - start_processing_time)
+        )
+        msg = ['Number of duplicates: {0}'.format(duplicates.value),
+               'Documents processed: ' + str(doc_len),
+               'Processing time: {0}'.format(total_time)]
```

Comment:
Optional comment: f-strings could be used

Reply:
Good catch! F-strings added